composite/synth-index/test/adapter.test.ts

```diff
@@ -0,0 +1,79 @@
+import { Requester } from '@chainlink/external-adapter'
+import { assertSuccess, assertError } from '@chainlink/adapter-test-helpers'
+import { AdapterRequest } from '@chainlink/types'
+import { makeExecute } from '../src/adapter'
+import { makeConfig } from '../src/config'
+
+describe('execute', () => {
+  const jobID = '1'
+  process.env.DATA_PROVIDER = 'coingecko'
+  const execute = makeExecute(makeConfig())
```

Comment:
Here you are using the env to inject the specific config into the underlying adapter. We should have the ability to provide the config directly and mock it here by writing an fn like `makeMockConfig`. 

`synth-index` fn needs to create the underlying `token-allocation` fn, so it should in its `makeConfig` probably require data to configure the underlying adapter.